src/H5FDcore.c

```diff
@@ -1708,3 +1710,28 @@ H5FD__core_unlock(H5FD_t *_file)
 done:
     FUNC_LEAVE_NOAPI(ret_value)
 } /* end H5FD__core_unlock() */
+
+/*-------------------------------------------------------------------------
+ * Function:    H5FD__core_delete
+ *
+ * Purpose:     Delete a file
+ *
+ * Return:      SUCCEED/FAIL
+ *
+ *-------------------------------------------------------------------------
+ */
+static herr_t
+H5FD__core_delete(const char *filename, hid_t H5_ATTR_UNUSED fapl_id)
+{
+    herr_t ret_value = SUCCEED; /* Return value             */
+
+    FUNC_ENTER_STATIC
+
+    HDassert(filename);
+
+    if (HDremove(filename) < 0)
```

Comment:
Seems like HDremove() should only be called when the 'backing store' flag is set in the FAPL.

Reply:
Fixed, though I think the behavior is awkward regardless.